_posts/2017-10-06-rust-blog-series-2.md

```diff
@@ -0,0 +1,1098 @@
+---
+layout: post
+title:  "Making a simple blog with Rust: Part II"
+date:   2017-10-06
+categories:
+- rust
+- programming
+- rocket
+- diesel
+---
+
+Welcome to Part II in our Rust blog hacking adventure.
+In this post, we'll be learning about connecting to the database
+and using some very simple Diesel APIs to seed that database.
+At the end of this post, we should have completed two fundamental tasks
+- Have a seeded database with users and posts,
+- Output each post title and author to our index page.
+
+## infer_schema!
+
+Our "entry point" for a Diesel applications.
+The Diesel [schema in depth guide] reviews all the options for database schema in greater details.
+All we need to know for the purpose of the guide is...
+
+>infer_schema! is a macro provided by diesel_codegen when you have enabled the feature for one or more database backends.
+The macro will establish a database connection at compile time,
+query for a list of all the tables,
+and generate infer_table_from_schema! for each one.
+infer_schema will skip any table names which start with __.
+
+Using `infer_schema!` will give us some implicit benefits later in development,
+but it's also very easy to get started with,
+which is why we're using it here.
```

Comment:
You talk about how this gives us benefits without talking about the possible alternatives